kratos/tests/test_model_part.py

```diff
@@ -659,6 +659,47 @@ def test_model_part_has_solution_step_variable(self):
 
         self.assertTrue(model_part.HasNodalSolutionStepVariable(VELOCITY))
         self.assertFalse(model_part.HasNodalSolutionStepVariable(PRESSURE))
+
+    def test_add_constraint(self):
+        model_part = ModelPart("Main")
+        model_part.AddNodalSolutionStepVariable(PRESSURE)
+        
+        n1 = model_part.CreateNewNode(1, 1.0,1.1,0.2)
+        n2 = model_part.CreateNewNode(2, 2.0,3.1,0.2)
+
+        VariableUtils().AddDof(PRESSURE,model_part)
+
+        c1 = MasterSlaveConstraint(10)
+        c2 = model_part.CeateNewMasterSlaveConstraint("UserProvidedLinearMasterSlaveConstraint", 1, n1, PRESSURE, n2, PRESSURE, 0.5, 0.0)

```

Comment:
c2 is not used anywhere.

Reply:
yes it is not. this is assigned as the function returns an object. I didn't want to leave the object unassigned.

Reply:
Ok, the tests could be improved by separating them into appropriately named test functions rather than mixing them in one function. Tests of functions intended to be used in c++ could be moved to the c++ test suite. Also, they could be more complete. For example, you create c2 with some values for the weight and constant. Why not test that CalculateLocalSystem returns the values expected?

Reply:
I created a dummy c-test to test exactly that
=> https://github.com/KratosMultiphysics/Kratos/blob/8bc6511025f1fb9455693b5b23baed5a5241b314/kratos/tests/sources/test_linear_master_slave_constraint.cpp

Reply:
This means this test will be heavy. That is a modelpart has to be created, a few nodes, variables and dofs. Will that be ok for the KratosCoreFastSuite ?

Reply:
this is done all the time, so its ok for the fastsuite

Reply:
Ideally, there would be one test per function. Each test should be minimal in the sense that it doesn't produce more work than is necessary to test that particular behavior. You can write a small test model factory to use in each of the unit tests.